machine_test.go

```diff
@@ -1772,6 +1821,329 @@ func TestCreateSnapshot(t *testing.T) {
 	}
 }
 
+func connectToVM(m *Machine, sshKeyPath string) (*ssh.Client, error) {
+	key, err := ioutil.ReadFile(sshKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         5 * time.Second,
+	}
+
+	if len(m.Cfg.NetworkInterfaces) == 0 {
+		return nil, errors.New("No network interfaces")
+	}
+
+	ip := m.Cfg.NetworkInterfaces.staticIPInterface().StaticConfiguration.IPConfiguration.IPAddr.IP
+
+	return ssh.Dial("tcp", fmt.Sprintf("%s:22", ip), config)
+}
+
+func writeCNIConfWithHostLocalSubnet(path, networkName, subnet string) error {
+	return ioutil.WriteFile(path, []byte(fmt.Sprintf(`{
+		"cniVersion": "0.3.1",
+		"name": "%s",
+		"plugins": [
+		  {
+			"type": "ptp",
+			"ipam": {
+			  "type": "host-local",
+			  "subnet": "%s"
+			}
+		  },
+		  {
+			"type": "tc-redirect-tap"
+		  }
+		]
+	  }`, networkName, subnet)), 0644)
+}
+
+func TestLoadSnapshot(t *testing.T) {
+	fctesting.RequiresKVM(t)
+	fctesting.RequiresRoot(t)
+
+	dir, err := ioutil.TempDir("", t.Name())
```

Comment:
[T.TempDir()](https://pkg.go.dev/testing#T.TempDir) ?

Reply:
It seems the entirety of machine_test.go uses ioutil.TempDir as opposed to its testing.T counterpart. So, I think it's fine to leave it as is.